### PR TITLE
ADEN-9185 Replace CMP modal with Fandom CMP

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: platforms/shared/bootstrap.ts [hunk 1]
```diff
@@ -1,13 +1,10 @@
-import { context, utils } from '@wikia/ad-engine';
+import { utils } from '@wikia/ad-engine';
 import { cmpWrapper } from './cmp/cmp-wrapper';
 
 export async function bootstrapAndGetCmpConsent(): Promise<boolean> {
 	const geo = utils.geoService.setUpGeoData().country;
```
Comment (fraszczakszymon) [inline platforms/shared/bootstrap.ts]:
```suggestion
	const countryCode = utils.geoService.setUpGeoData().country;
```

File: platforms/shared/cmp/cmp-wrapper.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
-import { utils } from '@wikia/ad-engine';
+import { context, utils } from '@wikia/ad-engine';
 import * as Cookies from 'js-cookie';
 
-const CMP_LIBRARY_URL = '//cdm.cursecdn.com/js/cmp/cmp-04-22/cmp.bundle.js';
-const optOutConsentString = 'BOhn5_-Ohn5_-ABABAENCVAAAAAn6AAA';
+const CMP_LIBRARY_URL = '//cdm.cursecdn.com/js/cmp/tracking-opt-in.min.js';
```
Comment (fraszczakszymon) [inline platforms/shared/cmp/cmp-wrapper.ts]:
```suggestion
const OPT_IN_MODAL_URL = '//cdm.cursecdn.com/js/cmp/tracking-opt-in.min.js';
```
?

File: platforms/shared/cmp/cmp-wrapper.ts [hunk 3]
```diff
@@ -19,100 +19,46 @@ export enum CmpPolicy {
...
+			// In case it fails to load, we'll resolve after 2s
 			setTimeout(() => {
+				utils.logger(logGroup, 'Timeout waiting for CMP to load');
 				resolve();
-			}, 1000);
+			}, 2000);
 
-			utils.scriptLoader.loadScript(CMP_LIBRARY_URL);
+			utils.scriptLoader.loadScript(CMP_LIBRARY_URL).then(() => {
+				utils.logger(logGroup, 'CMP Loaded');
```
Comment (fraszczakszymon) [inline platforms/shared/cmp/cmp-wrapper.ts]:
```suggestion
				utils.logger(logGroup, 'Modal library loaded');
```
?

File: platforms/shared/cmp/cmp-wrapper.ts [hunk 4]
```diff
@@ -19,100 +19,46 @@ export enum CmpPolicy {
  */
 class CmpWrapper {
```
Comment (fraszczakszymon) [inline platforms/shared/cmp/cmp-wrapper.ts]: It isn't CmpWrapper anymore, is it? It loads whole tracking-opt-in-modal, right?

File: platforms/shared/cmp/cmp-wrapper.ts [hunk 5]
```diff
@@ -203,58 +149,12 @@ class CmpWrapper {
...
-		window.__cmp.receiveMessage = cmpStored.receiveMessage;
-	}
-
 	/**
 	 * Returns whether a given geo requires GDPR consent
 	 *
 	 * @param geo The geo to check for GDPR consent
 	 */
-	geoRequiresConsent(geo: string): boolean {
+	private geoRequiresConsent(geo: string): boolean {
```
Comment (fraszczakszymon) [inline platforms/shared/cmp/cmp-wrapper.ts]: `tracking-opt-in` already has defined list of countries. Do we really need this method?

File: platforms/shared/bootstrap.ts (hunk 1 above)
Comment (dmnsobczak) [reply platforms/shared/bootstrap.ts]: Changed!

File: platforms/shared/cmp/cmp-wrapper.ts (hunk 2 above)
Comment (dmnsobczak) [reply platforms/shared/cmp/cmp-wrapper.ts]: Also changed.

File: platforms/shared/cmp/cmp-wrapper.ts (hunk 4 above)
Comment (dmnsobczak) [reply platforms/shared/cmp/cmp-wrapper.ts]: Not only. It also creates and switches dummy CMP before modal selection.

File: platforms/shared/cmp/cmp-wrapper.ts (hunk 3 above)
Comment (dmnsobczak) [reply platforms/shared/cmp/cmp-wrapper.ts]: Changed.

File: platforms/shared/cmp/cmp-wrapper.ts (hunk 5 above)
Comment (dmnsobczak) [reply platforms/shared/cmp/cmp-wrapper.ts]: Discussed offline and you right. Changed.

**Pull request merged**